### PR TITLE
add roadmap for 2023-24

### PR DESCRIPTION
**Pull request by nondiremanuel**
This is the blog post for 2023-24 roadmap.

### CODE REVIEW

**Comment by nondiremanuel** [comment]
This is an initial draft, please leave whatever comment or feedback.

The intended purpose is to let users know about the features included in BDK 1.0, divided in the different milestones, the expected timeline and to invite feedback.

File: docs/_blog/roadmap-2023.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    - Manuel Gatti
+    - Steve Myers
+date: "2023-10-17"
+tags: ["roadmap"]
+draft: false
+---
+
+Exciting times are ahead of the development of the Bitcoin Dev Kit, the rust-based set of libraries and tools that allow developers to seamlessly build cross-platform on-chain bitcoin wallets without having to re-implement standard bitcoin data structures, algorithms, and protocols. In the past few months great progress was made to bring new and improved BDK APIs to our users.
+
+As stated in a past [blog post](https://bitcoindevkit.org/blog/road-to-bdk-1/), the BDK team had temporarely frozen new features development, while keeping the maintenance of the 0.x versions ongoing, to focus on developing a new stable API - especially regarding the syncing mechanism, as brilliantly explained in the first [blog post](https://bitcoindevkit.org/blog/bdk-core-pt1/) regarding the new architecture of bdk_core. The goal of this major refactoring is to get to a BDK 1.0 version with a stable wallet API with new crates to manage the low level syncing blockchain data for a modern descriptor-based wallet with popular blockchain backends (e.g. Esplora, Electrum, RPC).
```
Comment (reez) [inline docs/_blog/roadmap-2023.md:14]: nit: "temporarely" -> "temporarily"

File: docs/_blog/roadmap-2023.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Timeline and Activities
+
+The graph below is a rough forcasted timeline for new development. Keep in mind the further the time from the present the higher the uncertainty of the estimations in case of number of contributors and possible changing priorities due to both technical and user requirements. 
+
+![](./bdk_roadmap_23-24.png)
+
+
+Only the macro-activities are listed in the graph and since it is not always easy to understand from the outside what they mean in practice, below you can also find a short description for each of major task (and a link to the relative PR on github when available).
```
Comment (reez) [inline docs/_blog/roadmap-2023.md:25]: nit: “github” -> “GitHub”

File: docs/_blog/roadmap-2023.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+Adjustment of the Esplora backend module due to the new way of storing blocks (see above), with a related new cli example showing off new chain syncing features.
+
+#### [Removal of TransactionDetails from Wallet API](https://github.com/bitcoindevkit/bdk/pull/1048)
+
+Change on the retrieval of details of a transaction such as amounts, fee and fee rate in order to make it directly possible from the Wallet API. 
+
+### 1.0 alpha.3 milestone
+
+This third milestone focuses on wrapping up Wallet API changes in preparation for the 1.0.0 beta release. It is composed by many small changes regarding the following main topics:
```
Comment (reez) [inline docs/_blog/roadmap-2023.md:66]: nit: “composed by” -> “composed of”

File: docs/_blog/roadmap-2023.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+The aim of this is to improve the readibility of different type of errors for the BDK users in order to be able to differentiate better in their own application.
+
+#### Further improving synching mechanism
+
+This includes creating a higher level method for syncing a Wallet using the new mechanisms recently added, improved the conflict resolution of unconfirmed transactions, some small bug fixes on the syncing mechanism and adding further examples for the different methodologies.
+
+### 1.0 beta release
+
+With the alpha.3 milestone underway, the plan is to release a BDK 1.0 beta release before the end of 2023. This BDK 1.0 beta version will provide a stable Wallet API our users can start experimenting with in their own application. During the beta phase the team will continue to add new code examples, documentation and testing. The team will also be updating the BDK-FFI mobile bindings (Kotlin/Android, Swift/iOS, python) to the new BDK 1.0 Wallet APIs (if you are more curious about BDK's general approach to bindings, this [this](https://bitcoindevkit.org/blog/bindings-scope/) blog post). New beta releases will be made as needed to address user feedback.
```
Comment (reez) [inline docs/_blog/roadmap-2023.md:78]: nit/suggestion: "this [this](https://bitcoindevkit.org/blog/bindings-scope/) blog post)" -> "check out [this](https://bitcoindevkit.org/blog/bindings-scope/) blog post)"

**Review by reez — COMMENTED** [review]
Incredibly well done! Really thorough yet concise read that gives great information. It's so good that the only suggestions from me are small little typo/grammatical/capitalization suggestions that I added a few comments for.

File: docs/_blog/roadmap-2023.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+### 1.0 release
+
+Once all beta user feedback, bug fixes, documentation, testing, and mobile language bindings are complete, the final BDK 1.0 will be ready to release. The purpose of calling this our 1.0 version is to signal to users that the main user facing APIs are stable and will be supported with bug fixes, documentation, and feature enhancements going forward. All projects using pre-1.0 versions of BDK should migrate to this version.
+
+### 1.1 release
+
+Current plans for a 1.0 release are still being developed but include plans to add compact block filter (BIP157/158) support and new electrum client features. No user facing Wallet APIs will be changed for this release, only new features will be added.
```
Comment (notmandatory) [inline docs/_blog/roadmap-2023.md:86]: I forgot that we also have some txbuilder fixes mentioned in bitcoindevkit/bdk#988 that wouldn't be breaking changes and could go in this release.

```suggestion
Current plans for a 1.0 release are still being developed but include plans to add compact block filter (BIP157/158) support and new electrum client features. We would also like to add some TxBuilder and coin selection enhancements. No user facing Wallet APIs will be changed for this release, only new features will be added.
```

**Comment by thunderbiscuit** [comment]
ACK 77dc259b060ff9e715411dc624fca0fa3d489d5b, given the comments above are addressed. How do you feel about just calling it the 2024 roadmap? At this point we might as well name it for the new year it's going to be merged in IMO.

**Comment by thunderbiscuit** [comment]
After discussion with the team, my understanding is that this is currently in holding and will get a big revamp and/or new post altogether. I'm on a journey to clean up the website PRs today so will close for now. Feel free to reopen it if I misunderstood!

**Pull request closed without merging**